Content/Items/Lightsaber/LightsaberBase.cs

```diff
@@ -0,0 +1,617 @@
+					break;
+				case ItemID.OrangePhaseblade:
+					item.shoot = ModContent.ProjectileType<LightsaberProj_Orange>();
+					break;
+			}
+
+			if (phaseblades.Contains(item.type))
+			{
+				item.damage = 16;
+				item.noUseGraphic = true;
+				item.noMelee = true;
+				item.autoReuse = false;
+				item.useTime = 12;
+				item.useAnimation = 12;
+				item.channel = true;
+				item.useStyle = ItemUseStyleID.Shoot;
+				item.knockBack = 2.5f;
+				item.shootSpeed = 14f;
+			}
+		}
+		public override bool AltFunctionUse(Item item, Player player)
+		{
+			if (!Main.Configuration.Get<bool>("Lightsabers", true))
+				return base.AltFunctionUse(item, player);
+
+			if (phaseblades.Contains(item.type) && item.type != ItemID.YellowPhaseblade && item.type != ItemID.OrangePhaseblade)
+				return true;
+
+			return base.AltFunctionUse(item, player);
+		}
+
+		public override bool Shoot(Item item, Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+		{
+			if (Main.Configuration.Get<bool>("Lightsabers", true))
+			{
+				if (item.type == ItemID.WhitePhaseblade && player.GetModPlayer<LightsaberPlayer>().whiteCooldown > 0 && player.altFunctionUse == 2)
+					return false;
+
+				if (phaseblades.Contains(item.type))
+				{
+					var proj = Projectile.NewProjectileDirect(source, position, velocity, type, damage, knockback, player.whoAmI);
+					(proj.ModProjectile as LightsaberProj).rightClicked = player.altFunctionUse == 2;
+					return false;
+				}
+			}
+
+			return base.Shoot(item, player, source, position, velocity, type, damage, knockback);
+		}
+	}
+
+	public abstract class LightsaberProj : ModProjectile
+	{
+		enum CurrentAttack : int
+		{
+			Slash1 = 0,
+			Slash2 = 1,
+			Slash3 = 2,
+			Slash4 = 3,
+			Throw = 4,
+			Reset = 5
+		}
+
+		private CurrentAttack currentAttack = CurrentAttack.Slash1;
+
+		protected bool frontHand = true;
+		protected bool updatePoints = true;
+
+		protected Vector2 oldScreenPos = Vector2.Zero;
+		protected Vector2 anchorPoint = Vector2.Zero;
+
+		protected int afterImageLength = 16;
+
+		protected int throwTimer = 0;
+		protected bool thrown = false;
+		protected bool turnedAround = false;
+		protected Vector2 thrownDirection = Vector2.Zero;
+
+		public bool rightClicked = false;
+		protected bool hide = true;
+		protected bool canHit = false;
+
+		protected bool initialized = false;
+
+		protected int attackDuration = 0;
+
+		protected float startRotation = 0f;
+
+		protected float midRotation = 0f;
+		protected float endMidRotation = 0f;
+		protected float startMidRotation = 0f;
+
+		protected float endRotation = 0f;
+
+		protected bool facingRight;
+
+		protected float squish = 1;
+
+		protected float startSquish = 0.4f;
+
+		protected float endSquish = 0.4f;
+
+		protected float rotVel = 0f;
+
+		protected int growCounter = 0;
+
+		protected List<float> oldRotation = new();
+		protected List<Vector2> oldPositionDrawing = new();
```

Comment:
same to above?

Reply:
I honestly don't know, but this is needed to prevent errors.